src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/MixedModeContextTests.cs

```diff
@@ -138,9 +138,11 @@ public override void RoundTripEmptyPoco()
             EmptyPoco expected = CreateEmptyPoco();
 
             string json = JsonSerializer.Serialize(expected, DefaultContext.EmptyPoco);
-            JsonTestHelper.AssertThrows_PropMetadataInit(() => JsonSerializer.Deserialize(json, DefaultContext.EmptyPoco), typeof(EmptyPoco));
+            // This would have thrown if we tried to lookup any properties but since there are no properties this is able to complete.
```

Comment:
Not sure I understand what has changed here, how has the refactoring impacted the change in functionality here?

Reply:
This is a bit tricky piece to explain as it tests partial initialization - JsonTypeInfo is initialized so that it can do serialization (it has set SerializeHandler) but it actually didn't provide info about properties so it cannot be deserialized for the most part. When I configure such type I mark it with "ThrowOnDeserialize".

The only place I check if I should be throwing on deserialize is when we look up for properties which is never called in the serializer. In this specific place we never lookup for properties because JSON `{}` doesn't have any properties and therefore deserialization can successfully finish. While I could do additional checks to make sure we preserve the old behavior I think it doesn't really make sense as we can actually do it.

Reply:
@layomia might want to provide feedback here.